### ISSUE TITLE
Incorrect Inspection Float + Int != Int

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#568]
Oops: https://youtrack.jetbrains.com/issue/WI-38293

```
/**
 * @param float $x
 */
function f($x) {
    // Incorrectly flagged with New value type (int) is not in annotated types
    $x = $x + 1;
}
```

PhpStorm 2017.3 EAP
Build #PS-173.2463.17, built on September 20, 2017
PhpStorm EAP User
Expiration date: October 20, 2017
JRE: 1.8.0_152-release-1008-b1 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.12.5

PHP Inspections Plugin: 2.3.10.6

**Comment by kalessil** [comment]
Duplicates #567

**Comment by kalessil** [comment]
Btw, huge thanks for crosss-posting @iakoder !

**Comment by iakoder** [comment]
I can almost be convinced of #567 but there's no way that a float + int could ever be considered anything other than a float.

var_dump(1.0+1);
double(2);

Is it possible that this error is coming from PhpStorm-provided type info?

```
/**
 * @param float $x
 *
 * @return float
 */
function f($x) {
    // On-hover type info on $t says "float|int"
    $t = $x + 1;
    return $t;
}
```

**Comment by kalessil** [comment]
On-hover displays what PhpStorm resolves as the type, so yes - this might be a bug.
